docs/installing-privateTemplate.md

```diff
@@ -0,0 +1,36 @@
+How to Fetch template from the private repository. In order to test the generator whether it fetch the template from the private repository or not, we will create a private repository in local and publish a template in the private repository.
+
+**Step 1:** Settuping Private Repository in Local:
+
+Installing [verdaccio](https://verdaccio.org/) using below command:
```

Comment:
I was thinking, wouldn't it be better to hide all of this and just extend https://github.com/asyncapi/generator/tree/master/test/test-project that is responsible for testing generator in isolated environment 🤔 

If I'm correct you would just have to:
- extend https://github.com/asyncapi/generator/blob/master/test/test-project/docker-compose.yml to handle startup of the verdaccio registory
  - here you have example https://verdaccio.org/docs/docker#using-docker-compose
  - you could add `htpasswd` file with username and password https://github.com/juanpicado/verdaccio-end-to-end-tests/blob/main/verdaccio/htpasswd and pass to it in docker compose command
  - you could also add packages that are available there by default https://github.com/juanpicado/verdaccio-end-to-end-tests/blob/main/verdaccio/verdaccio.yaml - that could be html-template that is already used in other tests (the specific version )

so with above you are getting rid of step 1-4 from this instruction

and to get rid last step:
- you would just add new file, like we already have `test-global.test.js`, that would do what you have in step 5, so programatically trigger generator and then evaluate if generation worked like a charm
- you would have to add a script in https://github.com/asyncapi/generator/blob/master/test/test-project/package.json#L5-L6 that would be fired after step 15 in https://github.com/asyncapi/generator/blob/master/test/test-project/test.sh

Thoughts? this way we get one document less to maintain, and out of the box good quality integration test that would work out of the box as we have GitHub Action workflow for it -> https://github.com/asyncapi/generator/blob/master/.github/workflows/pr-testing-with-test-project.yml
-

Reply:
> I was thinking, wouldn't it be better to hide all of this and just extend https://github.com/asyncapi/generator/tree/master/test/test-project that is responsible for testing generator in isolated environment 🤔
> 
> If I'm correct you would just have to:
> 
> * extend https://github.com/asyncapi/generator/blob/master/test/test-project/docker-compose.yml to handle startup of the verdaccio registory
>   
>   * here you have example https://verdaccio.org/docs/docker#using-docker-compose
>   * you could add `htpasswd` file with username and password https://github.com/juanpicado/verdaccio-end-to-end-tests/blob/main/verdaccio/htpasswd and pass to it in docker compose command
>   * you could also add packages that are available there by default https://github.com/juanpicado/verdaccio-end-to-end-tests/blob/main/verdaccio/verdaccio.yaml - that could be html-template that is already used in other tests (the specific version )
> 
> so with above you are getting rid of step 1-4 from this instruction
> 
> and to get rid last step:
> 
> * you would just add new file, like we already have `test-global.test.js`, that would do what you have in step 5, so programatically trigger generator and then evaluate if generation worked like a charm
> * you would have to add a script in https://github.com/asyncapi/generator/blob/master/test/test-project/package.json#L5-L6 that would be fired after step 15 in https://github.com/asyncapi/generator/blob/master/test/test-project/test.sh
> 
> ## Thoughts? this way we get one document less to maintain, and out of the box good quality integration test that would work out of the box as we have GitHub Action workflow for it -> https://github.com/asyncapi/generator/blob/master/.github/workflows/pr-testing-with-test-project.yml

@derberg I have tried using docker but we have to manually publish the template the repository, after providing the username and password in config, we have to still login, i have tried https://github.com/verdaccio/verdaccio/tree/master/docker-examples/v4/https-portal-example this example but also we have to manually publish the repo, Event the official video is publishing the repo manually [ We have to manually ]login https://www.youtube.com/watch?v=zRI0skF1f8I&t=234s&ab_channel=Docker 
<img width="1990" alt="image" src="https://github.com/asyncapi/generator/assets/60972989/3379a9fb-b834-493e-8cdf-969e94941fde">
* Also i cannot pass parameters in the cli for login and publishing the repository. 
<img width="1990" alt="image" src="https://github.com/asyncapi/generator/assets/60972989/cfbfa699-1b7d-4f55-a6bb-0ca642b76edf">

I guess existing the small doc is sufficient to help developers